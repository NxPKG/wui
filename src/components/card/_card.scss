// Start with a base of WuiPanel styles
@include wuiPanel($selector: '.wuiCard');

/**
 * 1. Footer is always at the bottom.
 * 2. Fix for IE where the image correctly resizes in width but doesn't collapse its height
      (https://github.com/philipwalton/flexbugs/issues/75#issuecomment-134702421)
 * 3. Horizontal layouts should always top left align no matter the textAlign prop
 */

// WuiCard specific
.wuiCard {
  display: flex;
  flex-direction: column;
  min-height: 1px; /* 2 */// sass-lint:disable-line mixins-before-declarations

  &.wuiCard-isDisabled {
    cursor: not-allowed; // duplicate property due to Chrome bug
    box-shadow: none;

    &:not(.wuiCard--plain) {
      background-color: $wuiColorLightestShade;
    }

    .wuiCard__title,
    .wuiCard__description,
    .wuiCard__top {
      color: $wuiColorDarkShade;
    }

    .wuiCard__top svg,
    .wuiCard__top img {
      filter: grayscale(100%);
    }

    .wuiCard__titleAnchor,
    .wuiCard__titleButton {
      cursor: inherit;
    }

    &:hover,
    &:focus {
      cursor: not-allowed; // duplicate property due to Chrome bug
      transform: none;
      box-shadow: none;
      text-decoration: none;
    }
  }

  &.wuiCard--isClickable {
    display: flex; // Override panel mixin

    &:not(.wuiCard-isDisabled) {
      // Progressive Enhancement Only
      // The title will still underline to simulate focus in IE
      &:focus-within {
        @include wuiSlightShadowHover;
        @include wuiFocusRing('large');
      }

      &:focus,
      &:hover {
        .wuiCard__title,
        .wuiCard__titleAnchor,
        .wuiCard__titleButton {
          text-decoration: underline;
        }
      }
    }
  }

  .wuiCard__top,
  .wuiCard__content,
  .wuiCard__footer {
    width: 100%; // Fixes IE and ensures they always stretch no matter the flex layout
  }

  &.wuiCard--leftAligned {
    text-align: left;
    align-items: flex-start;

    .wuiCard__titleButton {
      text-align: left;
    }
  }

  &.wuiCard--centerAligned {
    text-align: center;
    align-items: center;
  }

  &.wuiCard--rightAligned {
    text-align: right;
    align-items: flex-end;

    .wuiCard__titleButton {
      text-align: right;
    }
  }

  &.wuiCard-isSelected {
    transition: all $wuiAnimSpeedFast $wuiAnimSlightResistance;
  }
}

// This creates a bunch of sub selectors for the beta badge
@include wuiHasBetaBadge($selector: '.wuiCard', $spacing: $wuiCardSpacing);

.wuiCard--isSelectable {
  position: relative;
}

@each $modifier, $paddingAmount in $wuiCardPaddingModifiers {
  .wuiCard--#{$modifier} {
    padding: $paddingAmount;

    &.wuiCard--isSelectable {
      padding-bottom: $paddingAmount + $wuiCardBottomNodeHeight;
    }

    .wuiCard__top {

      .wuiCard__image {
        width: calc(100% + (#{$paddingAmount} * 2));
        left: $paddingAmount * -1;
        top: $paddingAmount * -1;
        margin-bottom: $paddingAmount * -1; // ensure the parent is only as tall as the image

        // IF both exist, position the icon centered on top of image
        + .wuiCard__icon {
          transform: translate(-50%, -75%); // Fallback for IE as it doesn't accept calcs in translates
          transform: translate(-50%, calc(-50% + #{$paddingAmount * -1})); // sass-lint:disable-line no-duplicate-properties
        }
      }
    }

  }
}

.wuiCard__top {
  flex-grow: 0; /* 1 */
  position: relative;
  min-height: 1px; /* 2 */

  .wuiCard__image {
    position: relative;

    // match border radius, minus 1px because it's inside a border
    border-top-left-radius: $wuiBorderRadius - 1px;
    border-top-right-radius: $wuiBorderRadius - 1px;
    overflow: hidden;

    img {
      width: 100%;
    }

    // IF both exist, position the icon centered on top of image
    + .wuiCard__icon {
      position: absolute;
      top: 50%;
      left: 50%;
    }
  }

  .wuiCard__icon {
    margin-top: $wuiCardSpacing / 2;
  }
}

.wuiCard__footer:not(:empty) {
  flex-grow: 0; /* 1 */
  margin-top: $wuiCardSpacing;
}

.wuiCard--hasChildren .wuiCard__description {
  margin-bottom: $wuiCardSpacing;
}

// Selectable cards don't work well without a border
.wuiCard--plain:not(.wuiCard--isSelectable) {
  border: none;
  box-shadow: none;

  .wuiCard__image {
    // Without a border, the image should stand on it's own so it needs to have all corners rounded
    border-radius: $wuiBorderRadius;
  }
}

@each $name, $color in $wuiCardSelectButtonBorders {
  .wuiCard--isSelectable--#{$name}.wuiCard-isSelected {
    border-color: $color;
  }
}

// If an icon or image exists, add some space
.wuiCard__top + .wuiCard__content {
  margin-top: $wuiCardSpacing;
}

.wuiCard__content {
  flex-grow: 1; /* 1 */

  .wuiCard__description {
    margin-top: $wuiCardSpacing / 2;
  }

  .wuiCard__titleAnchor,
  .wuiCard__titleButton {
    font: inherit;
    color: inherit;
    letter-spacing: inherit;

    &:focus {
      text-decoration: underline;
    }
  }

}

.wuiCard.wuiCard--horizontal {
  .wuiCard__content {
    text-align: left; /* 3 */
  }
}

// Only change the flex direction if the card has an icon
// otherwise the button tag won't properly align contents to top
.wuiCard.wuiCard--horizontal.wuiCard--hasIcon {
  flex-direction: row;
  // sass-lint:disable-block no-important
  align-items: flex-start !important; /* 3 */

  .wuiCard__top,
  .wuiCard__content {
    width: auto; // Makes sure the top shrinks and the content grows
    margin-top: 0;
  }

  .wuiCard__top .wuiCard__icon {
    margin-top: 0;
    margin-right: $wuiSize;
  }
}
