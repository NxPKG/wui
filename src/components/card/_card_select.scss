.wuiCardSelect {
  // Option select button that expands to sides and bottom
  @include wuiCardBottomNodePosition;
  font-weight: $wuiFontWeightBold;

  // Create button modifiers based upon the map.
  @each $name, $color in $wuiCardSelectButtonBackgrounds {
    &--#{$name}:enabled {
      background-color: $color;

      // Custom success text color since it doesn't exist on WuiButtonEmpty
      @if ($name == 'success') {
        color: makeHighContrastColor($wuiColorSuccess, $color);
      }
    }
  }

  &:disabled {
    background-color: $wuiPageBackgroundColor;
  }
}
