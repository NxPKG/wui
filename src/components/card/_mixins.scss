@mixin wuiCardBottomNodePosition {
  position: absolute;
  bottom: 0;
  left: 0;
  height: $wuiCardBottomNodeHeight !important;  // sass-lint:disable-line no-important -- To override .wuiButtonEmpty--xSmall
  width: 100%;
  overflow: hidden;
  // Subtract 1px from border radius since it sits inside a border-radius
  border-bottom-left-radius: $wuiBorderRadius - 1px;
  border-bottom-right-radius: $wuiBorderRadius - 1px;
}
