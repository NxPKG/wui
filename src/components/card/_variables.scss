$wuiCardSpacing: map-get($wuiPanelPaddingModifiers, 'paddingMedium');
$wuiCardBottomNodeHeight: $wuiSizeXXL;

$wuiCardSelectButtonBorders: (
  text: $wuiColorSuccess, // Default for selected
  primary: $wuiColorPrimary,
  success: $wuiColorSuccess,
  danger: $wuiColorDanger,
  ghost: $wuiColorDarkShade,
);

$wuiCardSelectButtonBackgrounds: (
  text: $wuiColorLightestShade, // Default for unselected
  primary: tintOrShade($wuiColorPrimary, 90%, 70%),
  success: tintOrShade($wuiColorSuccess, 90%, 70%),
  danger: tintOrShade($wuiColorDanger, 90%, 70%),
  ghost: $wuiColorDarkShade,
);

$wuiCardPaddingModifiers: (
  'paddingNone': 0,
  'paddingSmall': $wuiSizeS,
  'paddingMedium': $wuiSize,
  'paddingLarge': $wuiSizeL
) !default;