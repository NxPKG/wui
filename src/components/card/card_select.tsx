/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { FunctionComponent, ReactNode } from 'react';
import classNames from 'classnames';

import {
  WuiButtonEmpty,
  WuiButtonEmptyColor,
  WuiButtonEmptyProps,
} from '../button/button_empty';

import { WuiI18n } from '../i18n';

export type WuiCardSelectProps = WuiButtonEmptyProps & {
  /**
   * Is in the selected state
   */
  isSelected?: boolean;
  isDisabled?: boolean;
};

export const WuiCardSelect: FunctionComponent<WuiCardSelectProps> = ({
  className,
  isSelected = false,
  isDisabled,
  color,
  children,
  ...rest
}) => {
  const child = wuiCardSelectableText(isSelected, isDisabled, children);

  const selectClasses = classNames(
    'wuiCardSelect',
    `wuiCardSelect--${wuiCardSelectableColor(color, isSelected)}`,
    className
  );

  return (
    <WuiButtonEmpty
      className={selectClasses}
      color={color || 'text'}
      size="xs"
      isDisabled={isDisabled}
      iconType={isSelected ? 'check' : undefined}
      role="switch"
      aria-checked={isSelected}
      {...rest}>
      {child}
    </WuiButtonEmpty>
  );
};

function wuiCardSelectableText(
  isSelected: boolean | undefined,
  isDisabled: boolean | undefined,
  children: ReactNode
): ReactNode {
  if (children) {
    return children;
  }

  let text;

  if (isSelected) {
    text = <WuiI18n token="wuiCardSelect.selected" default="Selected" />;
  } else if (isDisabled) {
    text = <WuiI18n token="wuiCardSelect.unavailable" default="Unavailable" />;
  } else {
    text = <WuiI18n token="wuiCardSelect.select" default="Select" />;
  }

  return text;
}

export function wuiCardSelectableColor(
  color: WuiButtonEmptyColor | undefined,
  isSelected: boolean | undefined
): string {
  let calculatedColor;
  if (color) {
    calculatedColor = color;
  } else if (isSelected) {
    calculatedColor = 'success';
  } else {
    calculatedColor = 'text';
  }

  return calculatedColor;
}
