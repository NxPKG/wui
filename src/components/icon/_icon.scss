.wuiIcon {
  flex-shrink: 0; // Ensures it never scales down below its intended size
  display: inline-block;
  vertical-align: middle;
  fill: currentColor;
  transform: translate(0, 0); // Hack to fix Firefox "softness"

  &:focus {
    opacity: 1; // We often hide icons on hover. Make sure they appear on focus.
    background: $wuiFocusBackgroundColor;
  }
}

// App icons are two-toned so get an extra class. This provides the base color.
.wuiIcon--app {
  fill: $wuiTextColor;

  // This provides the default secondary color
  .wuiIcon__fillSecondary {
    fill: makeGraphicContrastColor($wuiColorPrimary);
  }
}

.wuiLogoFill {
  fill: $wuiWazuhLogoPrimaryColor;
}

.wuiLogoSecondaryFill {
  fill: $wuiWazuhLogoSecondaryColor;
}

.wuiIcon-isLoading {
  opacity: $wuiIconLoadingOpacity;
  background-color: currentColor;
  border-radius: $wuiBorderRadius;
}

.wuiIcon-isLoaded {
  animation: wuiIconLoading $wuiAnimSpeedNormal ease-in 0s 1 forwards;
}

// COLORS

@each $name, $color in $wuiIconColors {
  .wuiIcon--#{$name} {
    // Use color attribute so currentColor can be used in more properties
    color: $color;
  }
}

// App icon special cases for fully mono
.wuiIcon--text,
.wuiIcon--subdued,
.wuiIcon--primary {
  &,
  .wuiIcon__fillSecondary {
    fill: currentColor;
  }
}

// Use the negative fill on SVG elements that need to flip portions
// of the icon to light and dark compatible modes. You can see this
// in use on the Wazuh logo icons.
.wuiIcon__fillNegative {
  fill: $wuiColorDarkestShade;
}

// SIZES

@each $name, $size in $wuiIconSizes {
  .wuiIcon--#{$name} {
    @include size($size);
  }
}

@keyframes wuiIconLoading {
  0% {
    opacity: $wuiIconLoadingOpacity;
  }

  100% {
    opacity: 1;
  }
}
