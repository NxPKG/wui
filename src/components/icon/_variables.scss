$wuiIconLoadingOpacity: .05;

// Modifier naming and colors.
$wuiIconColors: (
  accent: makeGraphicContrastColor($wuiColorAccent),
  danger: makeGraphicContrastColor($wuiColorDanger),
  ghost: $wuiColorGhost,
  primary: makeGraphicContrastColor($wuiColorPrimary),
  secondary: makeGraphicContrastColor($wuiColorSecondary),
  success: makeGraphicContrastColor($wuiColorSuccess),
  subdued: $wuiTextSubduedColor,
  text: $wuiTextColor,
  warning: makeGraphicContrastColor($wuiColorWarning),
);

$wuiIconSizes: (
  small: $wuiSizeM,
  medium: $wuiSize,
  large: $wuiSizeL,
  xLarge: $wuiSizeXL,
  xxLarge: $wuiSizeXXL
);
