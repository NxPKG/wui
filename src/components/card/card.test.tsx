/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { requiredProps } from '../../test';

import { WuiCard, SIZES } from './card';

import { WuiIcon } from '../icon';

jest.mock('./../../services/accessibility', () => ({
  htmlIdGenerator: () => () => 'generated-id',
}));

describe('WuiCard', () => {
  test('is rendered', () => {
    const component = render(
      <WuiCard
        title="Card title"
        description="Card description"
        {...requiredProps}
      />
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    test('icon', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          icon={<WuiIcon className="myIconClass" type="apmApp" />}
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('a null icon', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          icon={null}
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('horizontal', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          layout="horizontal"
        />
      );

      expect(component).toMatchSnapshot();
    });

    describe('href', () => {
      it('supports href as a link', () => {
        const component = mount(
          <WuiCard title="Hoi" description="There" href="#" />
        );

        expect(component).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick as a link', () => {
        const handler = jest.fn();
        const component = mount(
          <WuiCard title="Hoi" description="There" href="#" onClick={handler} />
        );
        component.find('a').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });

      it('supports onClick as a button', () => {
        const handler = jest.fn();
        const component = mount(
          <WuiCard title="Hoi" description="There" onClick={handler} />
        );
        component.find('button').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });
    });

    test('titleElement', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          titleElement="h4"
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('titleSize', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          titleSize="xs"
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('footer', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          footer={<span>Footer</span>}
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('children', () => {
      const component = render(
        <WuiCard title="Card title" description="Card description">
          Child
        </WuiCard>
      );

      expect(component).toMatchSnapshot();
    });

    test('textAlign', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          textAlign="right"
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('display', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          display="plain"
        />
      );

      expect(component).toMatchSnapshot();
    });

    test('selectable', () => {
      const component = render(
        <WuiCard
          title="Card title"
          description="Card description"
          selectable={{
            onClick: () => {},
          }}
        />
      );

      expect(component).toMatchSnapshot();
    });

    describe('paddingSize', () => {
      SIZES.forEach(size => {
        test(`${size} is applied`, () => {
          const component = render(
            <WuiCard
              title="Card title"
              description="Card description"
              paddingSize={size}
            />
          );

          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
