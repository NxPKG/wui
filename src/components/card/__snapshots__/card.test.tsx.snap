// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WuiCard is rendered 1`] = `
<div
  aria-label="aria-label"
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props a null icon 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props children 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned wuiCard--hasChildren"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
    Child
  </div>
</div>
`;

exports[`WuiCard props display 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--plain wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props footer 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
  <div
    class="wuiCard__footer"
  >
    <span>
      Footer
    </span>
  </div>
</div>
`;

exports[`WuiCard props horizontal 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned wuiCard--horizontal"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props href supports href as a link 1`] = `
<WuiCard
  description="There"
  href="#"
  title="Hoi"
>
  <div
    className="wuiCard wuiCard--paddingMedium wuiCard--centerAligned wuiCard--isClickable"
    onClick={[Function]}
  >
    <div
      className="wuiCard__content"
    >
      <WuiTitle
        className="wuiCard__title"
        id="generated-idTitle"
        size="s"
      >
        <span
          className="wuiTitle wuiTitle--small wuiCard__title"
          id="generated-idTitle"
        >
          <a
            aria-describedby="generated-idDescription"
            className="wuiCard__titleAnchor"
            href="#"
            rel="noreferrer"
          >
            Hoi
          </a>
        </span>
      </WuiTitle>
      <WuiText
        className="wuiCard__description"
        id="generated-idDescription"
        size="s"
      >
        <div
          className="wuiText wuiText--small wuiCard__description"
          id="generated-idDescription"
        >
          <p>
            There
          </p>
        </div>
      </WuiText>
    </div>
  </div>
</WuiCard>
`;

exports[`WuiCard props icon 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned wuiCard--hasIcon"
>
  <div
    class="wuiCard__top"
  >
    <div
      class="myIconClass wuiCard__icon"
      data-wuiicon-type="apmApp"
    />
  </div>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props paddingSize l is applied 1`] = `
<div
  class="wuiCard wuiCard--paddingLarge wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props paddingSize m is applied 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props paddingSize none is applied 1`] = `
<div
  class="wuiCard wuiCard--paddingNone wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props paddingSize s is applied 1`] = `
<div
  class="wuiCard wuiCard--paddingSmall wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props selectable 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned wuiCard--isClickable wuiCard--isSelectable wuiCard--isSelectable--text"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
  <button
    aria-checked="false"
    aria-describedby="generated-idTitle generated-idDescription"
    class="wuiButtonEmpty wuiButtonEmpty--text wuiButtonEmpty--xSmall wuiCardSelect wuiCardSelect--text"
    role="switch"
    type="button"
  >
    <span
      class="wuiButtonContent wuiButtonEmpty__content"
    >
      <span
        class="wuiButtonEmpty__text"
      >
        Select
      </span>
    </span>
  </button>
</div>
`;

exports[`WuiCard props textAlign 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--rightAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props titleElement 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <h4
      class="wuiTitle wuiTitle--small wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </h4>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;

exports[`WuiCard props titleSize 1`] = `
<div
  class="wuiCard wuiCard--paddingMedium wuiCard--centerAligned"
>
  <div
    class="wuiCard__content"
  >
    <span
      class="wuiTitle wuiTitle--xsmall wuiCard__title"
      id="generated-idTitle"
    >
      Card title
    </span>
    <div
      class="wuiText wuiText--small wuiCard__description"
      id="generated-idDescription"
    >
      <p>
        Card description
      </p>
    </div>
  </div>
</div>
`;
